### PR TITLE
Back end/controllers/users.js

### PR DESCRIPTION
**Pull request by GundamMerlin**
Needs to be linted
please review

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -1,7 +1,94 @@
 const User = require('../models/user.js')
 const db = require('../db')
 const bcrypt = require("bcrypt");
-const jwt = require("jsonwebtoken")
+const jwt = require("jsonwebtoken");
+const { findByIdAndUpdate } = require('../models/user.js');
```
Comment (wjb108) [inline controllers/users.js]: think this needs be deleted

File: controllers/users.js [hunk 2]
```diff
@@ -1,7 +1,94 @@
...
+        res.status(500).json({error: error.message});
+    }
+};
+
+//GetUser
+const getUser = async (req, res) => {
+    try {
+        const { id } = req.params;
+        const user = await User.findById(id);s
+        res.status(200).json(user)
```
Comment (wjb108) [inline controllers/users.js]: try s character by error

File: controllers/users.js (hunk 1 above)
Comment (GundamMerlin) [reply controllers/users.js]: done

File: controllers/users.js (hunk 2 above)
Comment (GundamMerlin) [reply controllers/users.js]: done

Comment (wjb108) [inline controllers/users.js]: need a return before res

File: controllers/users.js [hunk 3]
```diff
@@ -1,7 +1,94 @@
...
+    } catch (error) {
+        res.status(500).json({error: error.message});
+    }
```
Comment (wjb108) [inline controllers/users.js:31]: return before res

File: controllers/users.js (hunk 2 above)
Comment (GundamMerlin) [reply controllers/users.js]: done

File: controllers/users.js (hunk 3 above)
Comment (GundamMerlin) [reply controllers/users.js:31]: done

File: controllers/users.js [hunk 4]
```diff
@@ -1,7 +1,94 @@
...
+};
+
+//UpdateUser
+
+const updateUser = async (req, res) => {
+    try {
+        const { id } = req.params;
+        const updatedUser = await findByIdAndUpdate (id, req.body, {new: true});
+        return res.status(200).json(updatedUser)
```
Comment (wjb108) [inline controllers/users.js:40]: add User. after await on line 40

File: controllers/users.js [hunk 5]
```diff
@@ -1,7 +1,94 @@
...
+    } catch (error) {
+        return res.status(500).json({error: error.message});
+    }
+};
+
+//DeleteUser
+
+const deleteUser = async (req,res) => {
+    try{
```
Comment (wjb108) [inline controllers/users.js:49]: check the repo for the deleteUser syntax and how they use deletedProduct variable and and if else statement with it
https://git.generalassemb.ly/sei-nyc-ducks/quacking-passwords/blob/master/controllers/products.js

File: controllers/users.js (hunk 4 above)
Comment (GundamMerlin) [reply controllers/users.js:40]: done

File: controllers/users.js (hunk 5 above)
Comment (GundamMerlin) [reply controllers/users.js:49]: done

**Review by foremanlb — APPROVED** [review]

**Pull request merged**